portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,34 +22,35 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns comment data in JSON format */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> comments = new ArrayList<String>();
+  static final ArrayList<String> comments = new ArrayList<String>();
```

Comment:
Sorry, I was a bit unclear! Given that you're always adding the same hard-coded values to the comments list, this suggests that you should just save the whole list as a constant instead of re-adding the same thing each time.

So something like:

private static final ArrayList<String> comments = Arrays.asList("hello", "world", "so cool!")

(which I believe should work, but it's been a while

You can also just static initialize an array here -- you're never modifying this list, so there's no need to use an ArrayList specifically.

Another thing is to make constants and instance fields private in most cases.